app/src/main/java/com/example/habitsmasher/ui/dashboard/HabitListFragment.java

```diff
@@ -148,47 +149,60 @@ private void initializeRecyclerView(LinearLayoutManager layoutManager, View view
         recyclerView.setLayoutManager(layoutManager);
         recyclerView.addItemDecoration(new DividerItemDecoration(getContext(), LinearLayoutManager.VERTICAL));
         recyclerView.setAdapter(_habitItemAdapter);
-        new ItemTouchHelper(_itemTouchHelperCallback).attachToRecyclerView(recyclerView);
-    }
 
-    /**
-     * The implementation of the swipe to delete functionality below came from the following URL:
-     * https://stackoverflow.com/questions/33985719/android-swipe-to-delete-recyclerview
-     *
-     *
-     * Name: Rahul Raina
-     * Date: November 2, 2016
-     */
-    // find a better way of doing this
-    ItemTouchHelper.SimpleCallback _itemTouchHelperCallback = new ItemTouchHelper.SimpleCallback(ItemTouchHelper.UP | ItemTouchHelper.DOWN, ItemTouchHelper.LEFT | ItemTouchHelper.RIGHT) {
-        @Override
-        public boolean onMove(@NonNull RecyclerView recyclerView,
-                              @NonNull RecyclerView.ViewHolder viewHolder,
-                              @NonNull RecyclerView.ViewHolder target) {
-            return false;
-        }
+        /* Implementation of swipe menu functionality came from this source:
+        Name: Velmurugan
+        Date: March 4, 2021
+        URL: https://howtodoandroid.com/android-recyclerview-swipe-menu
+         */
+        // create a touch listener which handles the swipe function of the RecyclerView
+        RecyclerTouchListener touchListener = new RecyclerTouchListener(getActivity(), recyclerView);
+        touchListener.setClickable(new RecyclerTouchListener.OnRowClickListener() {
+            @Override
+            // if row at the specified position is clicked
+            public void onRowClicked(int position) {
+                // Get the selected habit
+                Habit currentHabit = _habitItemAdapter._snapshots.get(position);
+
+                // Create a bundle to be passed into the habitViewFragment
+                Bundle bundle = new Bundle();
+                bundle.putSerializable("habit", currentHabit);
+                bundle.putSerializable("user", _user.getUsername());
+                NavController controller = NavHostFragment.findNavController(_fragment);
+
+                // Navigate to the habitViewFragment
+                controller.navigate(R.id.action_navigation_dashboard_to_habitViewFragment, bundle);
+            }
```

Comment:
New implementation of clicking Habit to view details using RecyclerTouchListener